code/game/objects/items/tools/mining_tools.dm

```diff
@@ -257,6 +250,9 @@
 		spark_system.set_up(5, 0, M)
 		spark_system.attach(M)
 		spark_system.start(M)
+		cell.charge += 200
+		to_chat(M, span_warning("[src] violently vaporizes chunks of the Xenomorph upon contact!"))
+		to_chat(user, span_notice("The vaporized matter causes a temporary feedback loop in the battery! <b>Charge Amount: [cell.charge]/[cell.maxcharge]</b>"))
```

Comment:
2 things: A) i can infinitely farm dead mobs for infinite charge
B) Vaporizes the xebnomorph but this can be used against any mob
C) use [] instead of the xenomorph

Reply:
Fixed all